soroban-sdk/src/error.rs

```diff
@@ -0,0 +1,33 @@
+use core::convert::Infallible;
+
+use crate::xdr;
+
+/// InvokeError captures errors returned from the invocation of another
+/// contract.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum InvokeError {
+    /// Abort occurs if the invoked contract function exited due to some
+    /// irrecoverable error, such as exceeding budget, or out-of-bounds access.
+    Abort,
+    /// Contract error occurs if the invoked contract function exited due to
```

Comment:
I would probably mention that this is an error that occurs when the contract explicitly returns an error or calls `panic_with_error`. I realize it's not necessarily in the scope of this enum, but I feel like it would be useful for the reader to make this connection.